### PR TITLE
ODIM-3824 error log doesn't have information about the supported and provided firmware version

### PR DESCRIPTION
**Pull request by irappa-pattar**

### CODE REVIEW

**Review by jeevan-kamkar — APPROVED** [review]
LGTM

**Review by tamanna-k — APPROVED** [review]

File: svc-aggregation/system/common.go [hunk 1]
```diff
@@ -1233,11 +1233,11 @@ func getConnectionMethodVariants(connectionMethodVariant string) connectionMetho
 	// Split the connectionmethodvariant and get the PluginType, PreferredAuthType, PluginID and FirmwareVersion.
 	// Example: Compute:BasicAuth:GRF_v1.0.0
 	cm := strings.Split(connectionMethodVariant, ":")
-	firmawareVesrion := strings.Split(cm[2], "_")
+	firmwareVesrion := strings.Split(cm[2], "_")
```
Comment (skhan-anjum) [inline svc-aggregation/system/common.go]: it should be "firmwareversion"

File: svc-aggregation/system/common.go [hunk 2]
```diff
@@ -1233,11 +1233,11 @@ func getConnectionMethodVariants(connectionMethodVariant string) connectionMetho
...
 	return connectionMethodVariants{
 		PluginType:        cm[0],
 		PreferredAuthType: cm[1],
 		PluginID:          cm[2],
-		FirmwareVersion:   firmawareVesrion[1],
+		FirmwareVersion:   firmwareVesrion[1],
```
Comment (skhan-anjum) [inline svc-aggregation/system/common.go]: it should be version instead of vesrion

File: svc-aggregation/system/common.go (hunk 1 above)
Comment (irappa-pattar) [reply svc-aggregation/system/common.go]: updated

File: svc-aggregation/system/common.go (hunk 2 above)
Comment (irappa-pattar) [reply svc-aggregation/system/common.go]: updated

**Review by a-ajith — APPROVED** [review]

**Review by skhan-anjum — APPROVED** [review]

**Review by amar-shalgar — APPROVED** [review]
LGTM

File: svc-aggregation/system/common.go [hunk 3]
```diff
@@ -1214,9 +1214,9 @@ func checkStatus(pluginContactRequest getResourceRequest, req AddResourceRequest
 		return common.GeneralError(http.StatusInternalServerError, response.InternalError, errMsg, nil, taskInfo), getResponse.StatusCode, queueList
 	}
 
-	// check the firmaware version of plugin is matched with connection method variant version
+	// check the firmware version of plugin is matched with connection method variant version
 	if statusResponse.Version != cmVariants.FirmwareVersion {
-		errMsg := "firmaware is not supported by connection method"
+		errMsg := fmt.Sprintf("Provided firmware version %s is not matched with supported firmware version %s of the plugin %s", cmVariants.FirmwareVersion, statusResponse.Version, cmVariants.PluginID)
```
Comment (Bharath-KKB) [inline svc-aggregation/system/common.go]: "Provided firmware version %s does not match supported firmware version %s of the plugin %s"

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
change as per comment

**Review by amar-shalgar — APPROVED** [review]
LGTM

File: svc-aggregation/system/common.go (hunk 3 above)
Comment (irappa-pattar) [reply svc-aggregation/system/common.go]: updated

**Review by Bharath-KKB — APPROVED** [review]

**Pull request merged**